### PR TITLE
Merge nectar

### PR DESCRIPTION
**Pull request by rishflab**
PR for merging nectar into the comit workspace. Nectars comit history has been preserved. The gitignore in nectar was moved into comit-rs. Nectar github CI config was removed and it uses the CI config from comit-rs.

### CODE REVIEW

**Comment by D4nte** [comment]
As I mentioned in the chat, I'd suggest to a PR on nectar for the formatting first.

Also, might be good to just ask for a `nectar freeze` when you get serious about this PR.

I suggest:
1. Merge pending Nectar PR
2. Announce Nectar freeze, get me or @thomaseizinger to archive nectar repo
3. proceed with this

**Comment by rishflab** [comment]
> As I mentioned in the chat, I'd suggest to a PR on nectar for the formatting first.
> 
> Also, might be good to just ask for a `nectar freeze` when you get serious about this PR.
> 
> I suggest:
> 
> 1. Merge pending Nectar PR
> 2. Announce Nectar freeze, get me or @thomaseizinger to archive nectar repo
> 3. proceed with this

Ill request the freeze.
Regarding the formatting, I was thinking it would be more efficient to let the formatting sort it self out after/during the merge as the nectar folder will become subject to the formatting rules in comit-rs.

**Comment by D4nte** [comment]
> > As I mentioned in the chat, I'd suggest to a PR on nectar for the formatting first.
> > Also, might be good to just ask for a `nectar freeze` when you get serious about this PR.
> > I suggest:
> > 
> > 1. Merge pending Nectar PR
> > 2. Announce Nectar freeze, get me or @thomaseizinger to archive nectar repo
> > 3. proceed with this
> 
> Ill request the freeze.
> Regarding the formatting, I was thinking it would be more efficient to let the formatting sort it self out after/during the merge as the nectar folder will become subject to the formatting rules in comit-rs.

I think it's best to minimize the changes during the merge and pre-emptively do them in nectar repo. It'll make review easier.

File: nectar/.geth_datadir/UTC--2020-06-24T07-35-41.322460345Z--88552c5d1a30ac899e3d17c63754f3f257e42663 [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{
```
Comment (thomaseizinger) [inline nectar/.geth_datadir/UTC--2020-06-24T07-35-41.322460345Z--88552c5d1a30ac899e3d17c63754f3f257e42663:1]: I would to know what this file is used for, seem redundant.

File: nectar/Cargo.lock [hunk 2]
```diff
@@ -0,0 +1,4468 @@
+# This file is automatically @generated by Cargo.
```
Comment (thomaseizinger) [inline nectar/Cargo.lock:1]: This lockfile should be removed now.

File: nectar/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+authors = ["CoBloX Team <team@coblox.tech>"]
+edition = "2018"
+
+[dependencies]
+anyhow = "1"
+async-trait = "0.1"
+bitcoin = { version = "0.23", features = ["rand", "use-serde"] }
+chrono = "0.4"
+clarity = "0.1"
+comit = { git = "https://github.com/comit-network/comit-rs/", package = "comit", rev = "76854090bb542ab55c5649c98ad1bc887bf5c764" }
```
Comment (thomaseizinger) [inline nectar/Cargo.toml:13]: This should be a path dependency now :)

File: nectar/bors.toml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+status = [
```
Comment (thomaseizinger) [inline nectar/bors.toml:1]: This file is redundant now.

File: nectar/.geth_datadir/UTC--2020-06-24T07-35-41.322460345Z--88552c5d1a30ac899e3d17c63754f3f257e42663 (hunk 1 above)
Comment (thomaseizinger) [reply nectar/.geth_datadir/UTC--2020-06-24T07-35-41.322460345Z--88552c5d1a30ac899e3d17c63754f3f257e42663:1]: @D4nte, @da-kami, @luckysori any insights?

File: Cargo.lock [hunk 5]
```diff
@@ -1,5 +1,20 @@
 # This file is automatically @generated by Cargo.
```
Comment (thomaseizinger) [inline Cargo.lock:1]: Quite a few dependencies being added here, we should audit which of those are actually needed :)

**Review by thomaseizinger — CHANGES_REQUESTED** [review]
Thanks for doing this @rishflab ! :)

There are a few more files that can be removed / changed!

File: nectar/Cargo.toml (hunk 3 above)
Comment (rishflab) [reply nectar/Cargo.toml:13]: was worried it wouldn't work when pointing to the latest comit but seems to be fine :)

File: .gitignore [hunk 6]
```diff
@@ -66,3 +66,14 @@ blockchain_nodes/bitcoin/
 blockchain_nodes/parity/parity
 .yalc
 yalc.lock
+
+# Default ignored files
```
Comment (D4nte) [inline .gitignore:70]: This kind of stuff can easily go in your home `.gitignore`

File: Cargo.lock (hunk 5 above)
Comment (rishflab) [reply Cargo.lock:1]: I updated some dependencies so they versions match which I can push but not sure how to do a proper audit

File: .gitignore (hunk 6 above)
Comment (thomaseizinger) [reply .gitignore:70]: I would assume this was just copy pasted from nectar though?

**Review by thomaseizinger — APPROVED** [review]

File: .gitignore (hunk 6 above)
Comment (rishflab) [reply .gitignore:70]: dropped that commit

Comment (rishflab) [reply .gitignore]: yep it was copy pasted from nectar @thomaseizinger

File: nectar/.geth_datadir/UTC--2020-06-24T07-35-41.322460345Z--88552c5d1a30ac899e3d17c63754f3f257e42663 (hunk 1 above)
Comment (D4nte) [reply nectar/.geth_datadir/UTC--2020-06-24T07-35-41.322460345Z--88552c5d1a30ac899e3d17c63754f3f257e42663:1]: I am running tests without it and see what happens ;)

**Review by D4nte — APPROVED** [review]
Did the following:
```
▶ cd ~/src/coblox/nectar
▶ find . -type f \! -path "*/.git/*" -exec shasum {} \; > ~/nectar_shashum
▶ cd ~/src/comit-network/comit-rs
▶ shasum -c ~/nectar_shashum|grep -v 'OK$'
./Cargo.toml: FAILED # Need to check
shasum: ./rust-toolchain: # Expected
./rust-toolchain: FAILED open or read
shasum: ./rustfmt.toml: No such file or directory  # Expected
./rustfmt.toml: FAILED open or read
shasum: ./LICENSE: No such file or directory # Expected
./LICENSE: FAILED open or read
shasum: ./Cargo.lock: No such file or directory # Expected
./Cargo.lock: FAILED open or read
shasum: ./.gitignore: No such file or directory # Expected
./.gitignore: FAILED open or read
shasum: ./.github/workflows/ci.yml: No such file or directory # Expected
shasum: ./bors.toml: No such file or directory # Expected
./.github/workflows/ci.yml: FAILED open or read
./bors.toml: FAILED open or read
shasum: WARNING: 7 listed files could not be read
shasum: WARNING: 1 computed checksum did NOT match
▶ diff ~/src/coblox/nectar/Cargo.toml ./Cargo.toml
13c13
< comit = { git = "https://github.com/comit-network/comit-rs/", package = "comit", rev = "76854090bb542ab55c5649c98ad1bc887bf5c764" }
---
> comit = { path = "../comit/", package = "comit" }
```

LGTM

File: Cargo.lock (hunk 5 above)
Comment (thomaseizinger) [reply Cargo.lock:1]: This tool is always a good start: https://github.com/est31/cargo-udeps

File: nectar/.geth_datadir/UTC--2020-06-24T07-35-41.322460345Z--88552c5d1a30ac899e3d17c63754f3f257e42663 (hunk 1 above)
Comment (D4nte) [reply nectar/.geth_datadir/UTC--2020-06-24T07-35-41.322460345Z--88552c5d1a30ac899e3d17c63754f3f257e42663:1]: I confirm it's needed for the tests :)
```
test swap::tests::execute_alice_hbit_herc20_swap ... FAILED

failures:

---- ethereum::geth::test::get_chain_id stdout ----
thread 'ethereum::geth::test::get_chain_id' panicked at 'called `Result::unwrap()` on an `Err` value: No such file or directory (os error 2)', src/ethereum/geth.rs:228:64

---- ethereum::wallet::tests::can_deploy_htlc stdout ----
thread 'ethereum::wallet::tests::can_deploy_htlc' panicked at 'called `Result::unwrap()` on an `Err` value: No such file or directory (os error 2)', src/ethereum/wallet.rs:567:55

---- ethereum::wallet::tests::ether_balance stdout ----
thread 'ethereum::wallet::tests::ether_balance' panicked at 'called `Result::unwrap()` on an `Err` value: No such file or directory (os error 2)', src/ethereum/wallet.rs:456:55

---- ethereum::wallet::tests::gas_limit stdout ----
thread 'ethereum::wallet::tests::gas_limit' panicked at 'called `Result::unwrap()` on an `Err` value: No such file or directory (os error 2)', src/ethereum/wallet.rs:488:55

---- ethereum::wallet::tests::gas_price stdout ----
thread 'ethereum::wallet::tests::gas_price' panicked at 'called `Result::unwrap()` on an `Err` value: No such file or directory (os error 2)', src/ethereum/wallet.rs:472:55

---- ethereum::wallet::tests::transfer_dai stdout ----
thread 'ethereum::wallet::tests::transfer_dai' panicked at 'called `Result::unwrap()` on an `Err` value: No such file or directory (os error 2)', src/ethereum/wallet.rs:512:55

---- swap::tests::execute_alice_hbit_herc20_swap stdout ----
Error: No such file or directory (os error 2)
thread 'swap::tests::execute_alice_hbit_herc20_swap' panicked at 'assertion failed: `(left == right)`
  left: `1`,
 right: `0`: the test returned a termination value with a non-zero status code (1) which indicates a failure', /Users/froyer/.rustup/toolchains/1.46.0-x86_64-apple-darwin/lib/rustlib/src/rust/src/libstd/macros.rs:16:9


failures:
    ethereum::geth::test::get_chain_id
    ethereum::wallet::tests::can_deploy_htlc
    ethereum::wallet::tests::ether_balance
    ethereum::wallet::tests::gas_limit
    ethereum::wallet::tests::gas_price
    ethereum::wallet::tests::transfer_dai
    swap::tests::execute_alice_hbit_herc20_swap

test result: FAILED. 140 passed; 7 failed; 7 ignored; 0 measured; 0 filtered out

error: test failed, to rerun pass '--bin nectar'
```

File: Cargo.lock (hunk 5 above)
Comment (D4nte) [reply Cargo.lock:1]: Can be done in follow-up PR please :)

**Pull request merged**